### PR TITLE
Remove banner dismissButtonLabel

### PR DESCRIPTION
**Pull request by msmithNI**
# Pull Request

## 🤨 Rationale

Follow-up tech debt from adding the label providers (#1090).

The banner's close button text can now be controlled by the `popupDismiss` label on `nimble-label-provider-core`, so the `dismissButtonLabel` property on `nimble-banner` is no longer needed. We decided to remove the property in [PR discussion on #1328](https://github.com/ni/nimble/pull/1328#discussion_r1248065530).

## 👩‍💻 Implementation

Remove property and update tests.

## 🧪 Testing

Updated autotests / Storybook.

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

**Comment by msmithNI** [comment]
@atmgrifter00 FYI added you as a buddy reviewer (though you'd be pulled in for the Blazor changes anyway)

**Review by atmgrifter00 — APPROVED** [review]

File: change/@ni-nimble-angular-4195d860-6f9d-49d1-83bc-7ccc1b77743d.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (rajsite) [inline change/@ni-nimble-angular-4195d860-6f9d-49d1-83bc-7ccc1b77743d.json:1]: Can you update the banner spec / quick double check the API it shows is mostly up to date?

**Review by rajsite — APPROVED** [review]

File: packages/nimble-components/src/banner/tests/banner.stories.ts [hunk 2]
```diff
@@ -125,12 +123,6 @@ export const _banner: StoryObj<BannerArgs> = {
             name: 'title-hidden',
             description: 'If set, hides the provided title.'
         },
```
Comment (jattasNI) [inline packages/nimble-components/src/banner/tests/banner.stories.ts:125]: For the number field story we kept a parameter called "localized labels" to link the client to the relevant label tokens for that component. Could we do the same for the banner?

**Review by jattasNI — APPROVED** [review]

File: change/@ni-nimble-angular-4195d860-6f9d-49d1-83bc-7ccc1b77743d.json (hunk 1 above)
Comment (msmithNI) [reply change/@ni-nimble-angular-4195d860-6f9d-49d1-83bc-7ccc1b77743d.json:1]: Done

File: packages/nimble-components/src/banner/tests/banner.stories.ts (hunk 2 above)
Comment (msmithNI) [reply packages/nimble-components/src/banner/tests/banner.stories.ts:125]: The banner Storybook page already has the `Localizable labels` section added from a previous PR.

**Pull request merged**